### ISSUE TITLE
Make it possible to boot a CIDER-compatible nREPL server via clj

### ISSUE THREAD

**Issue by bbatsov** [clojure-emacs/cider-nrepl#488]
Basically I want us to eliminate the needs for `lein/boot` to be installed if people just want to play with some Clojure without a project. (or if they are in a folder with just `deps.edn`

Basically we need either to add some `-main` to `cider-nrepl` or create a small wrapper package that would just start nREPL with our middleware handler. Something like 

```
clojure -Sdeps “{:deps {org.clojure/tools.nrepl {:mvn/version \“0.2.13\”} cider/cider-nrepl {:mvn/version \“0.17.0-SNAPSHOT\”}}}” -m cider.nrepl.run
```

See for context https://groups.google.com/forum/#!topic/clojure/RlHjnJPpFbU

**Comment by SevereOverfl0w** [comment]
I explored doing `-m cider.nrepl.run` but don't think this is a good idea. This makes it impossible for a user to provider additional switches, e.g. their own `-m`. I think `-e (cider.nrepl.run/go middleware to.inject)` is a better strategy, especially as multiple can be provided.

@bbatsov the context link is to your private inbox ;)

**Comment by bbatsov** [comment]
> @bbatsov the context link is to your private inbox ;)
`
Haha, seems I've messed up everything yesterday. :D I thought I was linking to Google Group's web UI. :D

> I explored doing -m cider.nrepl.run but don't think this is a good idea. This makes it impossible for a user to provider additional switches, e.g. their own -m. I think -e (cider.nrepl.run/go middleware to.inject) is a better strategy, especially as multiple can be provided.

Yeah, probably you're right. I'm completely fine with going down the `-e` route.

**Comment by bbatsov** [comment]
Fixed in #499.
